### PR TITLE
Improved label filter for tests

### PR DESCRIPTION
**Pull request by BetsyMcPhail**
The number of tests reported for a build can be incorrect when you're
filtering by label if some of the tests don't share a label with their
build.

Added a follow-up query when filtering by labels to determine how many
tests also have that label.

### CODE REVIEW

**Comment by BetsyMcPhail** [comment]
PTAL @zackgalbreath

File: public/api/v1/index.php [hunk 1]
```diff
@@ -1002,6 +1006,61 @@ function echo_main_dashboard_JSON($project_instance, $date)
...
+                            break;
+                        case 'failed':
+                            $test_response['nfaildiffp']++;
+                            break;
+                        case 'notrun':
+                            $test_response['nnotrundiffp']++;
+                            break;
+                    }
+                }
+            }
```
Comment (zackgalbreath) [inline public/api/v1/index.php]: The negative subscripts never get set here, but I'm okay with this (at least for now).  These numbers would be pretty difficult to compute based on how this data is currently stored.

Comment (BetsyMcPhail) [inline public/api/v1/index.php]: I agree. My thinking was that for now not being set is better than an incorrect value.

**Comment by zackgalbreath** [comment]
LGTM

**Pull request merged**